### PR TITLE
[BE] Provide access to the underlying module on NNModuleVariable, skip a string parsing calling to output_graph where possible

### PR DESCRIPTION
**Pull request by voznesenskym**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #114316

cc @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

**Comment by ezyang** [comment]
but it's failing tests?

File: torch/_dynamo/variables/nn_module.py [hunk 1]
```diff
@@ -74,12 +74,13 @@ def record_nn_module_stack(module_key: str, source, tx, mod: torch.nn.Module):
 
 
 class NNModuleVariable(VariableTracker):
-    _nonvar_fields = {"module_type", "module_key", *VariableTracker._nonvar_fields}
+    _nonvar_fields = {"module_type", "module_key", "module", *VariableTracker._nonvar_fields}
 
-    def __init__(self, module_type: type, module_key: str, **kwargs):
+    def __init__(self, module_type: type, module_key: str, module, **kwargs):
```
Comment (ezyang) [inline torch/_dynamo/variables/nn_module.py:79]: Is this guaranteed to be non-None?

**Comment by ezyang** [comment]
@jansel will have to review this, the old code seems to have been written in a very specific way and I am not sure why it was written that way

File: torch/_dynamo/variables/nn_module.py (hunk 1 above)
Comment (voznesenskym) [reply torch/_dynamo/variables/nn_module.py:79]: yes, everywhere this would be None should be a ConstantVariable(None)

**Comment by voznesenskym** [comment]
> but it's failing tests?

only lint?

**Review by jansel — CHANGES_REQUESTED** [review]
Why is this useful?  Seems like roughly the same amount of code, but now values are duplicated in multiple places which seems harder to maintain.

**Comment by voznesenskym** [comment]
> Why is this useful? Seems like roughly the same amount of code, but now values are duplicated in multiple places which seems harder to maintain.

Mild convenience of having to pass tx around, but I have another PR to make tx accessible anywhere, so i'll just go w/ that

**Comment by jansel** [comment]
We already have `InstructionTranslator.current_tx()`, no need to add something new.

**Pull request closed without merging**